android/src/main/java/io/branch/rnbranch/RNBranchModule.java

```diff
@@ -232,7 +207,18 @@ private void generateLocalBroadcast(JSONObject referringParams,
             }
         }.init(reactActivity);
 
-        branch.initSession(referralInitListener, uri, reactActivity);
+        notifyJSOfInitSessionStart(reactActivity, uri);
+        branch.sessionBuilder(reactActivity).withCallback(referralInitListener).withData(uri).init();
+    }
+
+    private static void notifyJSOfInitSessionStart(Context context, Uri uri) {
+        Intent broadcastIntent = new Intent(NATIVE_INIT_SESSION_STARTED_EVENT);
+        if (uri != null) {
+            broadcastIntent.putExtra(NATIVE_INIT_SESSION_STARTED_EVENT_URI, uri);
+        }
+
+        LocalBroadcastManager.getInstance(context).sendBroadcast(broadcastIntent);
+        Log.d(REACT_CLASS, "Sent session start broadcast for " + uri);
```

Comment:
Should prolly replace `Log.d` with `PrefHelper.Debug` so it is only printed when logging is enabled.

Reply:
:+1: Thanks for that suggestion.